app/views/admin/base.py

```diff
@@ -42,6 +72,12 @@ def setup(self, request, *args, **kwargs):
             for repo in sorted(model.Repo.list())
         ]
 
+    def get_params(self):
```

Comment:
Should each view override get_params() like this? It would be a bit tedious. How about making _GET_PARAMETERS and _POST_PARAMETERS methods, and let each view class only override them?:

```python
class MyView(...):
    def get_get_param_names(self):
        super(MyView, self).get_get_param_names() + ['param1', 'param2']
    def get(self, ...):
        params = self.get_params()  # Returns param1, param2 and inherited params
        ...
```

Reply:
That's a fair point. Though I personally feel more tedious with the current implementation where you always need to:

* Define MyView._GET_PARAMS
* Override get_params() which is mostly boilerplate

So you do need multiple definitions. But it is more like a personal preference, so I'm OK either ways.

But can you document in BaseView.get_params() how subclasses are supposed to override get_params(), maybe with sample code? It doesn't look obvious from the current document.

And optionally, maybe it's more concise by inlining the constants?:

```python
    def get_params(self):
        return views.base.read_params(
            super(AdminBaseView, self).get_params(),
            self.request,
            post_params={'xsrf_token': utils.strip})
```